.github/workflows/ci.yml

```diff
@@ -330,12 +329,51 @@ jobs:
           exit ${return}
       - name: Test
         run: |
-          (for DIM in 2 3; do \
-            printf "\n-------- Testing ${DIM}D --------\n"; \
-            cd ${{runner.workspace}}/build-${DIM}d-${{matrix.os}}-${{matrix.build_type}}; \
-            ctest ${{matrix.ctest_args}} -VV --output-on-failure; \
-            if [ $? -ne 0 ]; then exit 1; fi \
-          done)
+          cd ${{runner.workspace}}/build-${{matrix.dim}}d-${{matrix.os}}-${{matrix.build_type}}; \
+          ctest ${{matrix.ctest_args}} -VV --output-on-failure; \
+          if [ $? -ne 0 ]; then exit 1; fi \
+      - name: Checkout Reference Version
+        if: matrix.reference_comparison == 'ON'
+        run : |
+          git remote add amrexcomb https://github.com/AMReX-Combustion/PeleC.git
+          git fetch amrexcomb development
+          git checkout amrexcomb/development
+          git submodule update --recursive
+      - name: Configure Reference Version
+        if: matrix.reference_comparison == 'ON'
```

Comment:
Wondering if we should only do the reference check for `github.event_name == pull_request` too.

Reply:
Yeah it doesn't make sense to run the comparison with development after something has already been pushed into development.

Reply:
Done. I did `!= push` instead of `== pull_request` so that this will also run when the workflow is manually triggered so people can test out the comparison workflow before submitting a PR if they want to. 

I think this PR is ready to go now.